G/Graphviz/build_tarballs.jl

```diff
@@ -0,0 +1,57 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "Graphviz"
+version = v"2.42.3"
+
+# Collection of sources required to complete build
+sources = [
+    "https://www2.graphviz.org/Packages/stable/portable_source/graphviz-$(version).tar.gz" =>
+    "8faf3fc25317b1d15166205bf64c1b4aed55a8a6959dcabaa64dbad197e47add",
+
+    "./bundled",
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/graphviz-*/
+
+# Apply patch to build a native `mkdefs` utility that can be run within the
+# build environment.
+atomic_patch -p1 ../patches/gvpr-build-native-mkdefs.patch
+
+# This patch disable generation of dot's configuration
+atomic_patch -p1 ../patches/do-not-build-dot-config.patch
```

Comment:
`make install` would try to run the built `dot` binary executable in order to generate the configuration file.  This is a no-go within our environment as we can't run foreign executables apart from three platforms, so we necessarily have to skip this step.

The file could be generated by the Julia package using this JLL package, like [what we do in `Gtk.jl`](https://github.com/JuliaGraphics/Gtk.jl/blob/c64c641329abef189f67639e6989fff5993cc2f0/src/Gtk.jl#L84-L109), however the location of the configuration is tricky: this file must be in the same directory as the plugin libraries -- they're in the `lib/graphviz` directory of the tarball.

The location of this directory is determined by the `gvconfig_libdir` function in `lib/gvc/gvconfig.c` of the source code.  By default this directory should be something like `../lib/graphviz`, but this can be overridden by setting the environment variable `GVBINDIR`.  From [`doc` manual](https://www.graphviz.org/doc/info/command.html):

> `GVBINDIR`
>     Indicates which directory contains the Graphviz config file and plug-in libraries. If it is defined, the value overrides any other mechanism for finding this directory. If Graphviz is properly installed, it should not be needed, though it can be useful for relocation on platforms not running Linux or Windows.

I think we can generate the config file to a mutable artifact directory and copy there also the plugin libraries, then set `GVBINDIR` to this directory.

Reply:
*whistles* yeah i think that makes sense. Reallly great finds! :D

Your patch for building `mkdefs` is really cool. This stuff is neat! I'm impressed at how it all comes together! 😁